androidApp/src/main/java/com/mbta/tid/mbta_app/android/state/subscribeToVehicles.kt

```diff
@@ -0,0 +1,80 @@
+package com.mbta.tid.mbta_app.android.state
+
+import android.util.Log
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.asFlow
+import androidx.lifecycle.compose.LifecycleResumeEffect
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.mbta.tid.mbta_app.model.RouteDirection
+import com.mbta.tid.mbta_app.model.Vehicle
+import com.mbta.tid.mbta_app.model.response.ApiResult
+import com.mbta.tid.mbta_app.model.response.VehiclesStreamDataResponse
+import com.mbta.tid.mbta_app.repositories.IVehiclesRepository
+import org.koin.compose.koinInject
+
+class VehiclesViewModel(
+    private val vehiclesRepository: IVehiclesRepository,
```

Comment:
One thing worth mentioning here is that I didn't use the TImerViewModel that was used in subscribeToPredictions & subscribeToAlerts to synchronize updates. My perception was it was taking vehicles a long time to appear on the screen - perhaps because waiting for the 1s synchronized update. @JackVCurtis would be curious to learn more about your findings with using the TimerViewModel though - perhaps it should just have a more frequent rate.